faunadb-scala/src/test/scala/faunadb/ClientSpec.scala

```diff
@@ -919,6 +919,22 @@ class ClientSpec
       identifyR.to[Boolean].get shouldBe true
       }
 
+  it should "test HasCurrentToken" in {
+    val createR = client.query(Create(Collection("spells"), Obj("credentials" -> Obj("password" -> "abcdefg")))).futureValue
+
+    // Login
+    val loginR = client.query(Login(createR(RefField), Obj("password" -> "abcdefg"))).futureValue
+    val secret = loginR(SecretField).get
+
+    // HasCurrentToken Ok.
```

Comment:
Could we split this one into two separate test cases? I had to take a look at the ticket description to understand the correct and failing cases.

Something along the lines:

```
it should "return true when calling HasCurrentToken when logged in with an internal token" in
```

and

```
it should "return false when calling HasCurrentToken when logged in with an internal key" in
```

:)